src/lib/es2023.array.d.ts

```diff
@@ -0,0 +1,324 @@
+interface Array<T> {
+    /**
+     * Returns the value of the last element in the array where predicate is true, and undefined
+     * otherwise.
+     * @param predicate findLast calls predicate once for each element of the array, in descending
+     * order, until it finds one where predicate returns true. If such an element is found, findLast
+     * immediately returns that element value. Otherwise, findLast returns undefined.
+     * @param thisArg If provided, it will be used as the this value for each invocation of
+     * predicate. If it is not provided, undefined is used instead.
+     */
+    findLast<S extends T>(predicate: (this: void, value: T, index: number, obj: readonly T[]) => value is S, thisArg?: any): S | undefined;

```

Comment:
I don't think `this` should be typed `void`, that's probably an error in `es2015.core.d.ts`. I fixed it in my PR (#50450).
And I suggest repeating the doc too.

```suggestion
    findLast<S extends T>(predicate: (value: T, index: number, array: T[]) => value is S, thisArg?: any): S | undefined;
```

Reply:
FYI it’s now fixed by #51278.

Reply:
Thanks, I will fix the PR to match `.find()` declarations.